### PR TITLE
Duplicate submission notifications

### PR DESCRIPTION
**Pull request by daisieh**
Implements https://trello.com/c/ALce090M/36-feat-duplicate-submission-warnings-for-submitter-and-curator.

To test, create duplicate items with the same pubDOI or msid and see that you get a notification about the dup during the DescribePublicationStep and the OverviewStep. You should also see a list of duplicate submissions in any of the curator internal item views.

### CODE REVIEW

**Comment by ryscher** [comment]
Note: Must manually create the duplicateItem metadata field.

**Comment by daisieh** [comment]
What do you mean, must manually create the field?

File: dspace/modules/xmlui/src/main/java/org/dspace/app/xmlui/aspect/submission/submit/OverviewStep.java [hunk 1]
```diff
@@ -62,19 +63,24 @@
...
         Collection collection = submission.getCollection();
         String actionURL = contextPath + "/handle/" + collection.getHandle() + "/submit/" + knot.getId() + ".continue";
+        org.dspace.content.Item publication = DryadWorkflowUtils.getDataPackage(context, submission.getItem());
 
         body.addDivision("step-link","step-link").addPara(T_TRAIL);
+
         Division helpDivision = body.addDivision("general-help","general-help");
         helpDivision.setHead(T_MAIN_HEAD);
         helpDivision.addPara(T_MAIN_HELP);
+        if (publication.checkForDuplicateItems(context)) {
```
Comment (ryscher) [inline dspace/modules/xmlui/src/main/java/org/dspace/app/xmlui/aspect/submission/submit/OverviewStep.java]: I got a null pointer exception on this line. After submitting a duplicate item (same pub DOI), as a curator I claimed the task and attempted to edit the item metadata.

File: dspace/modules/xmlui/src/main/webapp/themes/Mirage/lib/css/style.css [hunk 2]
```diff
@@ -329,6 +329,12 @@ div.ds-option-set {
     box-shadow: 6px 6px 8px rgba(33, 33, 33, 0.25);
 }
 
+.duplicate-info {
+    background-color: #c4c4ff;
```
Comment (ryscher) [inline dspace/modules/xmlui/src/main/webapp/themes/Mirage/lib/css/style.css:333]: This color doesn't seem like an alert. Why not use the same colors as on the system alert box?  #ds-system-wide-alert, .notice.failure

**Comment by ryscher** [comment]
The field must either be manually entered into the metadata registry (as I did), or created using the metadata field update tool (which I forget the name of).

**Comment by ryscher** [comment]
Duplicate detection and alerting works. Code looks good. This can be accepted once the null pointer exception is fixed.

File: dspace/modules/xmlui/src/main/webapp/themes/Mirage/lib/css/style.css (hunk 2 above)
Comment (daisieh) [reply dspace/modules/xmlui/src/main/webapp/themes/Mirage/lib/css/style.css:333]: The system alert colors are difficult to notice if you're colorblind; I tested the various options using the Spectrum extension in Chrome and found that this was the highest-contrast. Honestly, we should switch all of them to something more like the purple.

**Comment by ryscher** [comment]
The null pointer issue is now fixed. 

New issue: Start a new submission. Select status "Published", and enter a DOI. This submission will be flagged as a duplicate, regardless of whether the DOI already exists in Dryad.

**Comment by ryscher** [comment]
Looks good. Merging.

**Pull request merged**